spec/sources/github_spec.rb

```diff
@@ -74,18 +73,51 @@ def fixture(file)
         message = @g.generate_github_description(warnings: [], errors:[])
         expect(message).to start_with("All green.")
       end
+
       it "Shows an error messages when there are errors" do
         message = @g.generate_github_description(warnings: [1,2,3], errors:[])
         expect(message).to eq("⚠ 3 Warnings. Don't worry, everything is fixable.")
       end
+
       it "Shows an error message when errors and warnings" do
         message = @g.generate_github_description(warnings: [1,2], errors:[1,2,3])
         expect(message).to eq("⚠ 3 Errors. 2 Warnings. Don't worry, everything is fixable.")
       end
+
       it "Deals with singualars in messages when errors and warnings" do
         message = @g.generate_github_description(warnings: [1], errors:[1])
         expect(message).to eq("⚠ 1 Error. 1 Warning. Don't worry, everything is fixable.")
       end
+    end
+
+    describe "issue creation" do
+      it "creates an issue if no danger comments exist" do
+        issues = []
+        allow(@g.client).to receive(:issue_comments).with("artsy/eigen", "800").and_return(issues)
+
+        body = @g.generate_comment(warnings: ["hi"], errors: [], messages: [])
+        expect(@g.client).to receive(:add_comment).with("artsy/eigen", "800", body).and_return({})
+
+        @g.update_pull_request!(warnings: ["hi"], errors: [], messages: [])
+      end
+
+      it "updates the issue if no danger comments exist" do
+        issues = [{:body => "generated_by_danger", :id => "12"}]
+        allow(@g.client).to receive(:issue_comments).with("artsy/eigen", "800").and_return(issues)
```

Comment:
Yes, and No. :dancer: Heh.

Yes, because it is part of the methods chain that are called. No, because that is not what we are testing here. 

To use@jonreid [here](https://speakerdeck.com/jonreid/unit-testing-control-your-dependencies?slide=6)

<img width="630" alt="screen shot 2016-01-10 at 08 10 40" src="https://cloud.githubusercontent.com/assets/49038/12220658/aca0b39e-b771-11e5-8702-a54ff541db92.png">

The stubbing of `issue_comments` is a part of the arrangement for the test, not the assertion that `:update_comment`gets called. That's why once these tests became longer, I started splitting it into three sections.